### ISSUE TITLE
pump-register-app prints stack traces instead of useful errors

### ISSUE THREAD

**Issue by ian-kelling** [pump-io/pump.io#1317]
see #1316 for example

**Comment by strugee** [comment]
Yeah, this is probably not so great. FWIW I'm guessing that particular stack trace was caused by the server returning some HTML (probably for a 404 or 500 response) instead of the JSON the client was expecting.
